src/main/java/org/aszjch/demoapp/aplication/ArticleFileController.java

```diff
@@ -0,0 +1,37 @@
+package org.aszjch.demoapp.aplication;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.domain.articlefile.ArticleFile;
+import org.aszjch.demoapp.domain.articlefile.ArticleFileService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import static org.springframework.http.HttpStatus.CREATED;
+
+@Slf4j
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/articles/files")
+class ArticleFileController {
+
+    private final ArticleFileService service;
+    private final ArticleFileDtoMapper mapper;
+
+    @PostMapping
+    ResponseEntity<String> uploadFile(@RequestParam Long articleId, @RequestPart MultipartFile uploadedFile) {
+        log.info("Creating articleFile");
+        ArticleFile articleFile = mapper.toEntity(articleId, uploadedFile);
```

Comment:
Jestem przyzwyczajony do dawania final wszędzie w parametrach i lokalnych zmiennych. Wtedy jak faktycznie jakąś zmieniasz w metodzie (np. bo masz pętle) to lepiej to widać. Zmień sobie ustawienia IntelliJ żeby Ci wstawiało finale :)